mobile/README.md

```diff
@@ -0,0 +1,127 @@
+- wip: integration with ByzCoin
+  - working: creating and updating credentials
+  - working: sending and receiving coins
+  - working: sample app to play rock-paper-scissors
+  - wanted: use personhood identity to add nodes to the byzcoin-network  
+- working: login services using the identity management
+- wanted: Peace on earth
+
+You use this at your own risk. It stores all your data unprotected on our test-network that is open to the public internet, 
+so be sure that you're OK with that!
+
+It is written in NativeScript and works on iOS and Android.
+
+## New V3 Version
+
+The old personhood.online available on the Google-play store uses an internal test-network of the ByzCoin
+blockchain. This new version is being ported to the V3 version of the ByzCoin blockchain, which is a more open, 
+polished, and used version of the DEDIS blockchain.
+
+## Bugs, comments
+
+Comments, suggestions are welcome either in the issues, PRs or via mail: <mailto:linus.gasser@epfl.ch>
+
+Please submit bugs and comments to the issue-tracker. Be sure to use the latest version of the software before
+submitting bugs.
+
+## Nativescript, crypto, and dynacred
+
+Using the `crypto` module in nativescript is not possible out-of-the box. This is why the app uses the
+`nativescript-nodeify` module. 
+Unfortunately this module is not supported anymore for the latest nativescript version, NS6.
+For this reason the app still uses NS5.
+Even with NS5 and `nativescript-nodeify`, there are some bugs that are not resolved yet.
+The `Makefile` has a `apply-patches` target that takes care of the following:
+- patch `nativescript-nodeify` to correctly recognize `@dedis/kyber`
+- link with dynacred in the repository
+
+So whenever the system gives you strange errors, a `make` should give you a good place to start again.
+Sorry for the inconvenience. 
+
+# Local Testing
+
+As of 22nd of July, the app is not completely ported to the V3 version of ByzCoin. If you want to test it
+locally and help debug it, here are the steps to do so.
+
+## Environment
+
+The tests suppose you have the following tools available:
+
+- npm == 6.9.0
+    - to check: `npm --version`
+    - to install on Mac: `brew install npm`
+
+- Node == 10.16.0
+    - to check: `node -v`
+    - to install on Mac: `brew install node@10.16`, then update your PATH so that `/usr/local/opt/node@10/bin` is at the front.
+
+- Mac: XCode >= 10.2
+    - command-line tools: `xcode-select --install`
+
+- NativeScript == 5.1.x
+    - to check: `tns --version`
+    - to install: `npm i -g nativescript@5.1`
+    
+- Docker
+    - to check - Mac: have the whale in your toolbar
+    - to install: https://download.docker.com/mac/stable/Docker.dmg  
+
+## Tests Overview
+
+For the tests, you need to run your own 4-node Cothority on your computer. The mobile app will then
+contact this Cothority.
+
+When in test mode, your app will contact this Cothority on the address 192.168.100.1, on ports 7771, 7773,
```

Comment:
Whut? Why don't use simply use `localhost`? That's overly weird to ask people to add a custom IP just to run tests (and every step on the paths to tests reduce the likeliness that theses get run).

Reply:
You cannot use localhost on a mobile device.
In fact, IIRC, android _does_ route requests to localhost to the hosting computer. But iOS doesn't. Or the other way round.

Reply:
Hum, that's really weird, "localhost" is just an alias in `/etc/hosts`, and (all?) libc resolve using it; maybe add an issue to check it latter?

Reply:
the app runs in the emulator...

Reply:
Then the emulator is started incorrectly, it should share the network with the host (as in docker network=host), or forward some ports.

Reply:
Wanna go and discuss it with the people from Apple?

It makes sense to separate the two: think of the emulator as it's own operating system, and it should get it's own IP address. And so it should also have it's own `localhost`...

Reply:
> Wanna go and discuss it with the people from Apple?

Sure, give me a mail :P

Or look at https://stackoverflow.com/questions/6077888/how-do-i-access-the-host-machine-itself-from-the-iphone-simulator

> It makes sense to separate the two: think of the emulator as it's own operating system, and it should get it's own IP address. And so it should also have it's own `localhost`...

It's using qemu behind the scenes, so, if you want full network virt, you get your own IP, if you want to share it with the host, you get the same IP.